core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -1414,7 +1415,9 @@ private static RelDataType multivalentStringWithSepSumPrecision(
     final RelDataType relDataType =
         typeFactory.getTypeSystem().deriveCovarType(typeFactory,
             opBinding.getOperandType(0), opBinding.getOperandType(1));
-    if (opBinding.getGroupCount() == 0 || opBinding.hasFilter()) {
+    if (opBinding.getGroupCount() == 0 || opBinding.hasFilter()
+        || opBinding.getOperator().kind == SqlKind.COVAR_POP
```

Comment:
are you sure about COVAR_POP?
Usually SAMP functions have the property that they return NULL for a non-empty set of size 1.

Reply:
According to https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/COVAR_POP.html. This function return type is nullable.
```
The function returns a value of type NUMBER. If the function is applied to an empty set, then it returns null.
```

Reply:
all aggregates return null when applied to an empty set.
but if getGroupCount() != 0 the set can never be empty (because then there would be no aggregate at all)

Reply:
this is about aggregates that are null even when the set is NOT empty

Reply:
empty set = set with all elements null

Reply:
I will do some test for this.